src/wp-includes/link-template.php

```diff
@@ -89,6 +89,56 @@ function permalink_anchor( $mode = 'id' ) {
 	}
 }
 
+/**
+ * Determine whether post should always use an ugly permalink structure.
+ *
+ * @since 5.7.0
+ *
+ * @param WP_Post|int|null $post   Optional. Post ID or post object. Defaults to global $post.
+ * @param bool             $sample Optional. Whether to force consideration based on sample links.
+ * @return bool Whether to use an ugly permalink structure.
+ */
+function wp_force_ugly_post_permalink( $post = null, $sample = null ) {
```

Comment:
Should this checking that the parent post also shouldn't have an ugly permalink? For instance if you have a public child with a private parent? Could probably just be a labeled a bad idea for a user to setup their posts like that, but thought I'd mention it.

Reply:
Oh good point. At the moment it's following the same rules as the `in_array()` checks it replaces but checking the post status object properties instead of hard-coding the names.

In the block editor, private pages don't show up as an option when searching for a parent page so it might be good for another ticket.

Reply:
Sounds good to me.